src/rtapi/uspace_rtapi_app.cc

```diff
@@ -977,11 +984,17 @@ int Posix::task_start(int task_id, unsigned long int period_nsec)
       return -errno;
   if(nprocs > 1) {
       const static int rt_cpu_number = find_rt_cpu_number();
-      cpu_set_t cpuset;
-      CPU_ZERO(&cpuset);
-      CPU_SET(rt_cpu_number, &cpuset);
-      if(pthread_attr_setaffinity_np(&attr, sizeof(cpuset), &cpuset) < 0)
-          return -errno;
+      if(rt_cpu_number != -1) {
+#ifdef __FreeBSD__
+          cpuset_t cpuset;
```

Comment:
If there are future instances of this needed in the future, may I suggest to put in one location
````
#ifdef __FreeBSD__
typedef cpuset_t rtapi_cpu_set_t
#else
typedef cpu_set_t rtapi_cpu_set_t
#endif

and use rtapi_cpu_set_t elsewhere, so that the ifdefs are kept to a minimum.